agent/agent.go

```diff
@@ -1481,8 +1481,10 @@ func (a *agent) createTailnet(
 	}()
 	if err = a.trackGoroutine(func() {
 		defer apiListener.Close()
+		apiHandler, closeAPIHAndler := a.apiHandler()
 		server := &http.Server{
-			Handler:           a.apiHandler(),
+			BaseContext:       func(net.Listener) context.Context { return ctx },
```

Comment:
Review: This change allows all `r.Context`s to inherit the graceful context from the agent, ensuring cancellation on agent shutdown.